plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -132,6 +143,79 @@ def get_token(module_params):
     }
 
 
+def is_struct_included(struct1, struct2, exclude=None):
+    """
+    This function compare if the first parameter structure is included in the second.
+    The function use every elements of struct1 and validates they are present in the struct2 structure.
+    The two structure does not need to be equals for that function to return true.
+    Each elements are compared recursively.
+    :param struct1:
+        type:
+            dict for the initial call, can be dict, list, bool, int or str for recursive calls
+        description:
+            reference structure
+    :param struct2:
+        type:
+            dict for the initial call, can be dict, list, bool, int or str for recursive calls
+        description:
+            structure to compare with first parameter.
+    :param exclude:
+        type:
+            list
+        description:
+            Key to exclude from the comparison.
+        default: None
+    :return:
+        type:
+            bool
+        description:
+            Return True if all element of dict 1 are present in dict 2, return false otherwise.
+    """
+    try:
+        if isinstance(struct1, list) and isinstance(struct2, list):
+            if len(struct1) == 0:
+                return True
+            found = False
+            for item1 in struct1:
+                found = False
```

Comment:
```suggestion
```

Reply:
I think we should keep the `found = False` on both line 
line 178: we need to assign it here because of the `return found` at the line 201 otherwise we would risk a reference before assignment error 
line 180: we want to check that every element of the struct1 is in struc2 so we have to set `found` at `false` on every iteration

Reply:
resolved by 4e5afb5